src/host/kvpair.rs

```diff
@@ -388,33 +435,8 @@ impl<const DEPTH: usize> MerkleTree<[u8; 32], DEPTH> for MongoMerkle<DEPTH> {
     }
 
     fn get_node_with_hash(&self, index: u32, hash: &[u8; 32]) -> Result<Self::Node, MerkleError> {
-        let v = self.get_record(index, hash).expect("Unexpected DB Error");
-        let height = (index + 1).ilog2();
-        v.map_or_else(
-            || {
-                let default = self.get_default_hash(height as usize)?;
-                let child_hash = if height == Self::height() as u32 {
-                    [0; 32]
-                } else {
-                    self.get_default_hash((height + 1) as usize)?
-                };
-                if default == *hash {
-                    Ok(MerkleRecord {
-                        index,
-                        hash: self.get_default_hash(height as usize)?,
-                        data: [0; 32],
-                        left: child_hash,
-                        right: child_hash,
-                    })
-                } else {
-                    Err(MerkleError::new(*hash, index, MerkleErrorCode::InvalidHash))
-                }
-            },
-            |x| {
-                assert!(x.index == index);
-                Ok(x)
-            },
-        )
+        let (node, _) = self.check_generate_default_node(index, hash)?;
```

Comment:
This function should not change.
As it should do the original functionality.
Just we currently do not use it in get_leaf_with_proof.